test/package/packageTest.nut.ts

```diff
@@ -0,0 +1,364 @@
+        version,
+        `'force:package:version:report' Version mismatch: expected '${version}', got '${result.Version}'`
+      );
+
+      expect(result.IsReleased, 'Expected IsReleased to be false').to.be.false;
+    });
+  });
+
+  describe.skip('install the package in scratch org', () => {
+    it('run force:package:install without waiting', () => {
+      const result = execCmd<{ Status: string; Id: string }>(
+        `force:package:install --json --package ${subscriberPkgVersionId} --installationkey ${INSTALLATION_KEY} --targetusername ${SUB_ORG_ALIAS} --publishwait 10`
+      ).jsonOutput.result;
+
+      expect(result).to.have.property('Status', 'IN_PROGRESS');
+      expect(result).to.have.property('Errors', null);
+      expect(result).to.have.property('SubscriberPackageVersionKey', subscriberPkgVersionId);
+      expect(result).to.have.property('Id');
+
+      installReqId = result.Id;
+    });
+
+    it('runs force:package:version:install:report until it finishes', async () => {
+      const waitForInstallRequestAndValidate = async (counter = 1): Promise<{ Status: string }> => {
+        const pollResult = execCmd<{ Status: string }>(
+          `force:package:install:report --requestid ${installReqId} --targetusername ${SUB_ORG_ALIAS} --json`
+        ).jsonOutput.result;
+
+        expect(pollResult).to.have.property('Status');
+        expect(pollResult).to.have.property('Id', installReqId);
+        expect(pollResult).to.have.property('Errors', null);
+        expect(pollResult).to.have.property('SubscriberPackageVersionKey', subscriberPkgVersionId);
+
+        if (pollResult.Status === 'IN_PROGRESS' && counter < 40) {
+          return sleep(WAIT_INTERVAL_MS, Duration.Unit.MILLISECONDS).then(() =>
+            waitForInstallRequestAndValidate(counter++)
+          );
+        } else {
+          // break out of recursion, validate pollResult
+          expect(
+            pollResult.Status,
+            `Checked InstallRequest ${counter} time(s) with interval of ${WAIT_INTERVAL_MS} ms and failed test with Status=${pollResult.Status}`
+          ).to.equal('SUCCESS');
+          return pollResult;
+        }
+      };
+      const result = await waitForInstallRequestAndValidate();
+      expect(result.Status).to.equal('SUCCESS');
+    });
+
+    it('run force:package:installed:list to verify in target scratch org', () => {
+      const result = execCmd<[{ SubscriberPackageVersionId: string }]>(
+        `force:package:installed:list --json --targetusername ${SUB_ORG_ALIAS}`
+      ).jsonOutput.result;
+
+      expect(result).to.have.length.at.least(1);
+      const foundRecord = result.filter((item) => item.SubscriberPackageVersionId === subscriberPkgVersionId);
+      expect(foundRecord, `Did not find SubscriberPackageVersionId ${subscriberPkgVersionId}`).to.have.length(1);
+      expect(foundRecord[0]).to.have.property('Id');
+    });
+  });
+
+  describe.skip('uninstall the package', () => {
```

Comment:
TODO: reenable all skipped tests

Reply:
Will re-enable as library includes each of the skipped functionality